src/azure-cli/requirements.txt

```diff
@@ -0,0 +1,167 @@
+azure-cli-monitor==0.2.8
+azure-cli-network==2.3.0
+azure-cli-nspkg==3.0.3
+azure-cli-policyinsights==0.1.0
+azure-cli-profile==2.1.2
+azure-cli-rdbms==0.3.5
+azure-cli-redis==0.3.2
+azure-cli-relay==0.1.2
+azure-cli-reservations==0.4.1
+azure-cli-resource==2.1.8
+azure-cli-role==2.3.0
+azure-cli-search==0.1.1
+azure-cli-servicebus==0.3.2
+azure-cli-servicefabric==0.1.11
+azure-cli-signalr==1.0.0
+azure-cli-sql==2.1.7
+azure-cli-storage==2.3.0
+azure-cli-telemetry==1.0.0
+azure-cli-vm==2.2.12
+azure-common==1.1.17
+azure-datalake-store==0.0.39
+azure-graphrbac==0.53.0
+azure-keyvault==1.1.0
+azure-mgmt-advisor==2.0.1
+azure-mgmt-authorization==0.50.0
+azure-mgmt-batch==5.0.1
+azure-mgmt-batchai==2.0.0
+azure-mgmt-billing==0.2.0
+azure-mgmt-botservice==0.1.0
+azure-mgmt-cdn==3.0.0
+azure-mgmt-cognitiveservices==3.0.0
+azure-mgmt-compute==4.4.0
+azure-mgmt-consumption==2.0.0
+azure-mgmt-containerinstance==1.2.1
+azure-mgmt-containerregistry==2.6.0
+azure-mgmt-containerservice==4.2.2
+azure-mgmt-cosmosdb==0.5.2
+azure-mgmt-datalake-analytics==0.2.0
+azure-mgmt-datalake-nspkg==3.0.1
+azure-mgmt-datalake-store==0.5.0
+azure-mgmt-datamigration==0.1.0
+azure-mgmt-devtestlabs==2.2.0
+azure-mgmt-dns==2.1.0
+azure-mgmt-eventgrid==0.4.0
+azure-mgmt-eventhub==2.2.0
+azure-mgmt-hdinsight==0.2.0
+azure-mgmt-iotcentral==1.0.0
+azure-mgmt-iothub==0.6.0
+azure-mgmt-iothubprovisioningservices==0.2.0
+azure-mgmt-keyvault==1.1.0
+azure-mgmt-loganalytics==0.2.0
+azure-mgmt-managementgroups==0.1.0
+azure-mgmt-maps==0.1.0
+azure-mgmt-marketplaceordering==0.1.0
+azure-mgmt-media==1.0.1
+azure-mgmt-monitor==0.5.2
+azure-mgmt-msi==0.2.0
+azure-mgmt-network==2.4.0
+azure-mgmt-nspkg==3.0.2
+azure-mgmt-policyinsights==0.1.0
+azure-mgmt-rdbms==1.5.0
+azure-mgmt-recoveryservices==0.1.0
+azure-mgmt-recoveryservicesbackup==0.1.1
+azure-mgmt-redis==5.0.0
+azure-mgmt-relay==0.1.0
+azure-mgmt-reservations==0.3.1
+azure-mgmt-resource==2.0.0
+azure-mgmt-search==2.0.0
+azure-mgmt-servicebus==0.5.3
+azure-mgmt-servicefabric==0.2.0
+azure-mgmt-signalr==0.1.1
+azure-mgmt-sql==0.11.0
+azure-mgmt-storage==3.1.1
+azure-mgmt-trafficmanager==0.51.0
+azure-mgmt-web==0.40.0
+azure-multiapi-storage==0.2.2
+azure-nspkg==3.0.2
+azure-storage-blob==1.3.1
+azure-storage-common==1.4.0
+azure-storage-nspkg==3.1.0
+bcrypt==3.1.6
+certifi==2018.11.29
+cffi==1.11.5
+chardet==3.0.4
+colorama==0.4.1
+cryptography==2.5
+entrypoints==0.3
+futures==3.1.1
+humanfriendly==4.17
+idna==2.8
+ipaddress==1.0.22
+isodate==0.6.0
+jmespath==0.9.3
+keyring==17.1.1
+knack==0.5.1
+mock==2.0.0
+msrest==0.6.4
+msrestazure==0.4.34
+oauthlib==3.0.1
+paramiko==2.4.2
+pbr==5.1.1
+portalocker==1.2.1
+prompt-toolkit==1.0.15
+pyasn1==0.4.5
+pycparser==2.19
+pydocumentdb==2.3.3
+pygments==2.3.1
+pyjwt==1.7.1
+pynacl==1.3.0
+pyopenssl==19.0.0
+python-dateutil==2.7.5
+pytz==2018.9
+pyyaml==4.2b4
+requests-oauthlib==1.2.0
+requests==2.21.0
+scp==0.13.0
+six==1.12.0
+sshtunnel==0.1.4
+tabulate==0.8.2
+urllib3[secure]==1.24.1
```

Comment:
This is where its debatable, because that's an indirect dependencies. But assuming "azure-cli" is an application and not an SDK, pinning exactly the tested version is the only to have reproducible installation in the future: when I install azure-cli 2.5.78 released on March 2021, I get the exact same environment.

Reply:
I think we are more concerned that you get the exact version of all the CLI modules, not every specific dependency.

Reply:
@tjprescott @marstr If that's the case, then pin everything "azure*", that solves all the current problem we have, and avoid the Python version complicated story.